pkg/apis/monitoring/v1/types.go

```diff
@@ -1344,7 +1347,7 @@ type PodMonitorSpec struct {
 	// Per-scrape limit on length of labels value that will be accepted for a sample.
 	// Only valid in Prometheus versions 2.27.0 and newer.
 	LabelValueLengthLimit uint64 `json:"labelValueLengthLimit,omitempty"`
-	// Attaches node metadata to discovered targets. Only valid for role: pod.
+	// Attaches node metadata to discovered targets. Only valid for role: pod and service.
```

Comment:
I think that the `Only valid for role: pod ...` part isn't really needed here because by definition, a PodMonitor configures the pod's role.

```suggestion
	// Attaches node metadata to discovered targets.
```